ui/src/common/useSelectedFields.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { useDispatch, useSelector } from 'react-redux'
+import { setOmittedFields, setToggleableFields } from '../actions'
+
+const useSelectedFields = ({ resource, columns, omittedColumns = [] }) => {
+  const dispatch = useDispatch()
+  const resourceFields = useSelector(
+    (state) => state.settings.toggleableFields
+  )?.[resource]
+  const omittedFields = useSelector((state) => state.settings.omittedFields)?.[
+    resource
+  ]
+
+  const [filteredComponents, setFilteredComponents] = useState([])
+
+  useEffect(() => {
+    if (
+      !resourceFields ||
+      Object.keys(resourceFields).length !== Object.keys(columns).length
+    ) {
+      const obj = {}
+      for (const key of Object.keys(columns)) {
+        obj[key] = true
+      }
+      dispatch(setToggleableFields({ [resource]: obj }))
+    }
+    if (!omittedFields) {
+      dispatch(setOmittedFields({ [resource]: omittedColumns }))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
If in the future ND, uses some libraries for rehydrating redux store after new releases like [redux-persist](https://www.npmjs.com/package/redux-persist), then this useEffect could be skipped then.